scipy/ndimage/tests/test_filters.py

```diff
@@ -137,17 +159,17 @@ def _validate_complex(self, array, kernel, type2, mode='reflect', cval=0):
             convolve(array, kernel, output=real_dtype)
 
         # raises if output array is provided, but is not complex-valued
-        output_real = np.zeros_like(array, dtype=real_dtype)
+        output_real = xp.zeros_like(array, dtype=real_dtype)
         with assert_raises(RuntimeError):
             correlate(array, kernel, output=output_real)
 
         with assert_raises(RuntimeError):
             convolve(array, kernel, output=output_real)
 
-    def test_correlate01(self):
-        array = np.array([1, 2])
-        weights = np.array([2])
-        expected = [2, 4]
+    def test_correlate01(self, xp):
+        array = xp.array([1, 2])
```

Comment:
`xp.array` isn't part of the standard, but this is OK because this is only for CuPy?

Reply:
I would rather we didn't do piecemeal conversions though - if we are already writing `xp.`..., we should go straight to the standard where we can.

Reply:
Yup, that's an oversight of search-and-replace. Will fix